contracts/nexus-gateway/src/contract/execute.rs

```diff
@@ -20,18 +57,115 @@ mod test {
 
     use connection_router::state::{CrossChainId, Message};
 
+    use crate::state::{Config, MockStore};
+
     use super::*;
 
+    #[test]
+    fn test_route_messages_unauthorized() {
```

Comment:
we don't have to start test functions with `test_`. They already have the `#[test]` attribute and from what I've seen so far it seems to be convention to directly describe the test, e.g. `do_not_route_messages_from_unauthorized_sender`